### DISCUSSION TITLE
The actual use of file_pool_size

### DISCUSSION THREAD

**Discussion by DJPlaya** [arvidn/libtorrent#6439] [Q&A] [answered]
Hello,
Ime just fine tuning my high performance torrent client, and i was wondering what is the actual usage of the file_pool_size setting nowadays.
I am using Sophos on my internal Proxy VM and have an UTM set to max security.
The server has Sophos Endpoint, it should skip most networked files since it communicates with the UTM, so basically there is no big need for local virus file checks.

In the documentation, it is just described as useful for virus scanners, so my main question is if there is any good use in having a high number set in the settings for performance. If ime not completely mistaken, the limit just shouldent be lower then the global active torrent limit, right?

**Comment by arvidn** [comment, marked as answer]
> In the documentation, it is just described as useful for virus scanners, so my main question is if there is any good use in having a high number set in the settings for performance.

The reason there is a pool is to avoid high costs to `open` and `close` calls. Both of those can become very expensive with virus scanners, but also on MacOS with HFS+, `close()` can be extremely slow. In more recent versions of libtorrent, it's not just `open` and `close`, but also `mmap` and `munmap` calls that are saved by caching them in the pool. I'm less familiar with how kernels normally handle frequent mapping and unmapping, but there's a risk each call have to alter the page table and could be somewhat expensive.

> If ime not completely mistaken, the limit just shouldent be lower then the global active torrent limit, right?

The size of the file pool is not really related to the number of torrents you have. It needs to be set below the per-process file descriptor limit (with margin to allow for peer connections, since those are also file descriptors). As for a lower limit, the more different files are actively read from or written to, the higher you want the pool size to be in order to get the most effect.
